juju/juju
# fix 1749300: don't save backup to db by default

## Description of change

juju create-backup should only download a backup by default, not save a copy to the db also.  Added a new flag, --keep-copy, to allow for a copy to be saved to the db at creation time.  Since there is a 
no-download flag currently, it will now imply keep-copy, as you must do one or the other.  

## QA steps

1. juju bootstrap localhost testme
2. juju switch controller
3. juju create-backup
3. verify compressed download file downloaded.
4. juju backups, 
4. results should show no backup ids.
6. on controller, verify there are no /tmp/juju-backups directories
7. juju create-backup --keep-copy
8. verify compressed file downloaded.
4. juju backups
5. one backup id should be listed, same as printed during create-backup
6. juju create-backup --no-download
4. verify no new compressed backup file downloaded.
5. juju backups
4. results should show another backup id

## Documentation changes

Yes, 

## Bug reference

https://bugs.launchpad.net/juju/+bug/1749300
